zkvm/src/types.rs

```diff
@@ -216,6 +228,15 @@ impl Data {
         }
     }
 
+    /// Downcast the data item to a `CallProof` type.
+    pub fn to_call_proof(self) -> Result<CallProof, VMError> {
```

Comment:
per comment above - we can probably parse the callproof on prover also, so no need for this method.